### PR TITLE
404: drop pug in favor of jsx

### PR DESCRIPTION
**Pull request by flootr**
Part of _Eliminate Pug_ (#19822). Removes the Pug.js template which is used for the 404 response.

### Testing

Compare the sources (Right-click > "View page source") of https://wpcalypso.wordpress.com/asdf with the corresponding output of this branch (https://calypso.live/asdf?branch=update/404-template). There should not be any significant differences.

### CODE REVIEW

File: client/document/404.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Head from '../components/head';
+
+class NotFound extends Component {
+	render() {
```
Comment (ockham) [inline client/document/404.jsx]: If the component only has a `render` method, we should turn it into a `PureComponent`, or better yet, a stateless functional one 😬

**Comment by ockham** [comment]
One more thing that just occurred to me: Can we use the [`<EmptyContent />`](https://github.com/Automattic/wp-calypso/tree/master/client/components/empty-content) component here? The JSX in 404.jsx is essentially mimicking that component, so it'd be better to re-use the existing component 😁

**Comment by samouri** [comment]
code lgtm, i'll test now

**Comment by flootr** [comment]
Thanks @samouri for bringing this up 👍 I'm not really sure which page is served there but I don't think it's the one which got served by Calypso.

For example https://wpcalypso.wordpress.com/asdf is showing the same design as this PR. I guess there's some site in front of Calypso which is handling the 404 instead?

**Comment by ockham** [comment]
> Thanks @samouri for bringing this up 👍 I'm not really sure which page is served there but I don't think it's the one which got served by Calypso.
>
> For example https://wpcalypso.wordpress.com/asdf is showing the same design as this PR. I guess there's some site in front of Calypso which is handling the 404 instead?

That's correct, wordpress.com routes have to be whitelisted in our nginx config, whereas wpcalypso.wordpress.com ones don't. As a result, there's that non-Calypso 404 on wordpress.com that @samouri found.

Arguably, not a lot of wordpress.com users will end up seeing the Calypso 404, but it still makes sense to have it -- both for non-production/staging environments, and for when we might direct all wordpress.com routes to Calypso.

**Comment by ockham** [comment]
(So to be clear, this PR isn't going to change the production/staging wordpress.com 404 and thus shouldn't need designer approval.)

**Comment by sirreal** [comment]
You can see the production Calypso 404 at https://wordpress.com/log-in/show-me-a-404 - better for comparison.

**Comment by samouri** [comment]
Whoops, sorry for holding this up based on my misunderstanding!
Thanks for sharing that knowledge @ockham!

🚢 away!

**Comment by flootr** [comment]
Thanks @ockham @sirreal and @samouri for the valuable feedback 👍

**Pull request merged**